psiminer-core/src/test/kotlin/BasePsiRequiredTest.kt

```diff
@@ -42,17 +46,27 @@ open class BasePsiRequiredTest : BasePlatformTestCase() {
     private class UnknownMethodException(methodName: String, fileName: String) :
         RuntimeException("Can't find method $methodName in $fileName")
 
-    protected fun getJavaMethod(methodName: String): PsiElement =
-        javaMethods[methodName] ?: throw UnknownMethodException(methodName, javaMethodsFileName)
-
-    protected fun getKotlinMethod(methodName: String): PsiElement =
-        kotlinMethods[methodName] as KtFunction
+    fun getMethod(methodName: String): PsiElement =
+        methods[methodName] ?: throw UnknownMethodException(methodName, psiSourceFile.name)
 
     companion object {
-        // We can not get the root of the class resources automatically
+        // We cannot get the root of the class resources automatically.
         private const val resourcesRoot: String = "data"
+    }
+}
 
-        internal const val javaMethodsFileName = "JavaMethods.java"
-        internal const val kotlinMethodsFileName = "KotlinMethods.kt"
+open class JavaPsiRequiredTest(source: String) : BasePsiRequiredTest(dataFolder.resolve("$source.$ext")) {
+    override val handler: LanguageHandler = JavaHandler()
+    companion object {
+        val dataFolder = File("java")
+        const val ext = "java"
+    }
+}
+
+open class KotlinPsiRequiredTest(source: String) : BasePsiRequiredTest(dataFolder.resolve("$source.$ext")) {
```

Comment:
Why this classes are not abstract?

Reply:
And why they should be since they don't define any abstract methods?

Reply:
I just thought that it was not intended that the user would create instances of these classes, but only inherit and write their tests inside the inherited classes. Abstract modifier is perfect for this.

Reply:
It's already something strange if the user creates an instance of a test class. By this logic, we should mark all test classes with an abstract modifier.